src/org/javarosa/xform/util/XFormUtils.java

```diff
@@ -78,7 +103,12 @@ public static FormDef getFormFromInputStream(InputStream is) throws XFormParseEx
                 throw new XFormParseException("IO Exception during parse! " + uee.getMessage());
             }
 
-            return _factory.getXFormParser(isr).parse();
+            XFormParser xFormParser = _factory.getXFormParser(isr);
+            xFormParser.setExternalInstancePathPrefix(externalInstancePathPrefix);
```

Comment:
I understand that this line (107) is the only one needed to fix the issue, right? (see comments about the `ParserCustomizer` class)